apps/mason2/vcf_materialization.cpp

```diff
@@ -214,6 +214,9 @@ void VcfMaterializer::init()
                 throw MasonIOException("Could not write methylation levels FAI index.");
         }
     }
+
+    if (!empty(vcfFileName) && numSeqs(faiIndex) != length(contigNames(context(vcfFileIn))))
+        throw MasonIOException("Number of contigs in FAI index and VCF differ.");
```

Comment:
```suggestion
    {
        throw MasonIOException("Number of contigs in the reference FAI index and VCF file differ. "
                                "Make sure your VCF file is based on reference file " + std::string{CString(fastaFileName)});
    }
```
Not sure if the cast works as intended

Reply:
I used a stringstream to not cast myself into despair :)

Looks like this now:
```
____INITIALIZING______________________________________________________________

Opening reference and variants file ...[...]/build/seqan/include/seqan/basic/basic_exception.h:363 FAILED!  (Uncaught exception of type MasonIOException: The Number of contigs in FAI index and VCF differ:
    195 contig(s) in FAI Index [...]/GCA_000001405.15_GRCh38_no_alt_analysis_set.fasta.fai
    1 contig(s) in VCF file [...]/HG002_21.vcf
Ensure the VCF file and FASTA reference contain the same contigs. Either by supplying entries for each reference in the VCF file, or by removing all contigs from the FASTA reference that are not part of the VCF.)
```